server.js

```diff
@@ -16,9 +25,104 @@ const app = express();
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
+// API documentation
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  try {
+    const endpoints = expressListEndpoints(app).map((endpoint) => {
+      if (endpoint.path === "/thoughts") {
+        endpoint.query = {
+          description: `Here you can send a GET request to get the last 20 thoughts in a decending order. As well as posting a new thought to the API with a POST request. Please include a JSON with the message like that: { "message": "My Happy Thought" }`,
+        };
+      }
+      if (endpoint.path === "/thoughts/:thoughtId/like") {
+        endpoint.query = {
+          description:
+            "On this endpoint you can like a thought by sending a POST request. Please provide the Id of the desired thought for it to work.",
+        };
+      }
+      return endpoint;
+    });
+    res.status(200).json(endpoints);
+  } catch (error) {
+    console.error("The following error occured:", error);
+    res
+      .status(500)
+      .send(
+        "Sorry, this page is not available at the moment. Please try again later."
+      );
+  }
+});
+
+//GET the thoughts endpoint
+//Also sorting them in a decending order and limit the thoughts to 20 shown
+app.get("/thoughts", async (req, res) => {
+  try {
+    const allThoughts = await Thought.find()
+      .sort({ createdAt: -1 })
+      .limit(20)
+      .exec();
+
+    if (allThoughts.length > 0) {
+      res.status(200).json(allThoughts);
+    } else {
+      res.status(404).send("No Thoughts were found.");
+    }
+  } catch (error) {
+    console.error("The following error occured:", error);
+    res
+      .status(500)
+      .send(
+        "Sorry, this page is not available at the moment. Please try again later."
+      );
+  }
+});
+
+//POST the thoughts endpoint
+app.post("/thoughts", async (req, res) => {
+  const { message, hearts, createdAt, _id } = req.body;
```

Comment:
do we need to provide "_id" for posting a thought? I don't think we need to extract that from request body as MongoDB will automatically assign that to new data.